### PR TITLE
Improve pull request description template

### PR DESCRIPTION
**Pull request by oneonestar**
## Description

Copy from https://github.com/trinodb/trino/pull/23636 with the following changes:
* Remove the change on checkbox (Looks like the checkbox still being converted into a tasklist)
* Update links
```diff
-at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md 
-and contact us on #core-dev in Slack. -->
+at https://trinodb.github.io/trino-gateway/development/#contributing
+and contact us on #trino-gateway-dev in Slack. -->
```

## Release notes

- (x) This is not user-visible or is docs only, and no release notes are required.
- ( ) Release notes are required. Please propose a release note for me.
- ( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by mosabua — APPROVED** [review]
Looks good to me.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
+<!-- Mark the appropriate option with an (x). Propose a release note if you can.
+More info at https://trino.io/development/process#release-note -->
 ## Release notes
 
-( ) This is not user-visible or is docs only, and no release notes are required.
-( ) Release notes are required. Please propose a release note for me.
-( ) Release notes are required, with the following suggested text:
+- ( ) This is not user-visible or is docs only, and no release notes are 
+  required.
```
Comment (ebyhr) [inline .github/pull_request_template.md:20]: Unwarp.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+- ( ) Release notes are required. Please propose a release note for me.
+- ( ) Release notes are required, with the following suggested text:
 
 ```markdown
-* 
+## Section
```
Comment (ebyhr) [inline .github/pull_request_template.md:25]: I don't think "Section" is helpful in this project.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,21 +1,27 @@
-<!-- Thank you for submitting a pull request! --> 
-<!-- Provide an overview of the PR for maintainers and reviewers. -->
+<!-- Thank you for submitting a pull request! Find more information
+at https://trino.io/development/process.html,
+at https://trinodb.github.io/trino-gateway/development/#contributing
```
Comment (ebyhr) [inline .github/pull_request_template.md:3]: It's hard to copy these link. Ideally, it shouldn't have a leading `at ` or a following `,`.

Comment (mosabua) [reply .github/pull_request_template.md:3]: I was talking to @wendigo in trino PR .. maybe we change these all to markdown outside the comment and at the end of the template.. then the links render properly. Wdyt @ebyhr

Comment (mosabua) [reply .github/pull_request_template.md:3]: I can send a follow up PR here or in Trino repo

File: .github/pull_request_template.md (hunk 1 above)
Comment (ebyhr) [reply .github/pull_request_template.md:20]: https://github.com/trinodb/trino-gateway/pull/510

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
-( ) Release notes are required, with the following suggested text:
+- ( ) This is not user-visible or is docs only, and no release notes are 
+  required.
+- ( ) Release notes are required. Please propose a release note for me.
+- ( ) Release notes are required, with the following suggested text:
 
 ```markdown
-* 
+## Section
+* Fix some things. ({issue}`issuenumber`)
```
Comment (ebyhr) [inline .github/pull_request_template.md:26]: This format doesn't work well in this project. The existing release note format is 
```
* xxx. ([#prnumber](https://github.com/trinodb/trino-gateway/pull/prnumber))
or
* xxx. ([#issuenumber](https://github.com/trinodb/trino-gateway/issue/issuenumber))
```

**Pull request merged**